File: easy/1957. Delete Characters to Make Fancy String/1957. Delete Characters to Make Fancy String.ts
function makeFancyString(s: string): string {
    // const res = s.slice(0, 2).split("")
    let res = ""
    let prevOne = res[0]
    let prevTwo = res[1]

    for (const sym of s) {
        if (prevOne === prevTwo && prevOne === sym) {
            continue
        }

        prevOne = prevTwo
        prevTwo = sym
        res += sym
    }

    return res
};

File: easy/1957. Delete Characters to Make Fancy String/1957. Delete Characters to Make Fancy String.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        if len(s) == 1:
            return s

        res = list(s)[:2]
        prev = res[-1]
        prev_two = res[-2]

        for sym in s[2:]:
            if prev == prev_two == sym:
                continue

            prev_two = prev
            prev = sym
            res.append(sym)

        return "".join(res)
